pkg/webhook/pod/pod.go

```diff
@@ -65,13 +68,34 @@ func (a *podAnnotator) Handle(ctx context.Context, req admission.Request) admiss
 		return admission.Errored(http.StatusInternalServerError, err)
 	}
 
-	marshaledPod, _ := json.Marshal(pod)
+	// admission.PatchResponse generates a Response containing patches.
+	return patchPodChanges(originalPod, pod)
+}
+
+// create jsonpatches only to changed caused by the kubemacpool webhook changes
+func patchPodChanges(originalPod, currentPod *corev1.Pod) admission.Response {
```

Comment:
I think I can make a good reuse of `patchChange()` [function](https://github.com/k8snetworkplumbingwg/kubemacpool/pull/259/files#diff-276495201a043c6a12276502c3cf1a1db02fd2c7c9f4323b20415c49b244e9dbR125), but not sure where I should place it...